Tests for forbidden host code-points and resolution of non-special URLs

alwinb opened the pull request:
This adds additional tests for:

1. The forbidden host codepoints.
2. Resolution of non-special URLs, 
e.g. `abc:rootless` against `abc://host/path` should result in `abc:rootless`.

Related issues:

- https://github.com/whatwg/url/issues/458
- https://github.com/whatwg/url/pull/589
- https://github.com/alwinb/spec-url/issues/6

/cc @TimothyGu 

**Edit** It seems some of the forbidden host code points were already tested in https://github.com/web-platform-tests/wpt/pull/23572.
Let me know if I should remove the (few) duplicates.

alwinb committed:
Tests for forbidden host code-points and resolution of non-special URLs

TimothyGu commented on url/resources/urltestdata.json:
```diff
@@ -7794,5 +7794,204 @@
     "username": ""
+  },
+  "Forbidden host codepoints",
+  { "input": "foo://ho\u0000st/",
```
Would it be possible to add these tests near the tests added in #23572?

Also, the prevailing style in the file is to have `{` not be on the same line as the first property.

TimothyGu commented on url/resources/urltestdata.json:
```diff
@@ -7794,5 +7794,204 @@
+    "base": "about:blank",
+    "failure": true
+  },
+  "Ensure that non-special URLs use strict resolution",
```
"strict resolution" isn't a term used in the URL Standard. Let's just say

```suggestion
  "Ensure that schemes are not dropped when resolving non-special URLs",
```

domenic reviewed (commented):
Normative content of tests LGTM, but it'd be great to address the style/naming issues pointed out by @TimothyGu.

alwinb committed:
Move tests up; slight formatting changes; rephrase comment

alwinb commented:
I understand and agree, no need to point it out twice. 

I moved the entire block of forbidden host codepoints up and removed the three duplicates. 
I went with a just slightly different wording for the comment, I think it is a bit more clear. 
I changed the formatting and also found one other line starting with `{ "`. Of my hand I am afraid, apologies. I changed that one as well.
